### PR TITLE
feat: add `partition_xml` for XML files

### LINKED ISSUE #595: Add a `partition_xml` function for XML documents
The goal of this issue is to add a `partition_xml` function for extracting text from XML documents.

### PR DESCRIPTION
**Pull request by MthwRobinson**
### Summary

Closes #595. Adds a `partition_xml` function for processing XML files. If `keep_xml_tags=True`, XML tag information is included in the elements. Otherwise, only text attributes are returned. `keep_xml_tags=False` be default.

### Testing

```python
from unstructured.partition.xml import partition_xml

elements = partition_xml(filename="example-docs/factbook.xml", keep_xml_tags=True)

elements = partition_xml(filename="example-docs/factbook.xml", keep_xml_tags=False)
```

### CODE REVIEW

**Review by qued — APPROVED** [review]
LGTM, one suggestion and a few typos. One correction on the metadata type but I went ahead and fixed it.

File: unstructured/partition/auto.py [hunk 1]
```diff
@@ -43,6 +44,7 @@ def partition(
     ssl_verify: bool = True,
     ocr_languages: str = "eng",
     pdf_infer_table_structure: bool = False,
+    keep_xml_tags: bool = False,
```
[the remark this answers is not in the document]
Comment (MthwRobinson) [reply unstructured/partition/auto.py:47]: Yep that sounds good to me.

File: unstructured/partition/auto.py [hunk 2]
```diff
@@ -83,6 +85,9 @@ def partition(
         additional metadata field, "text_as_html," where the value (string) is a just a
         transformation of the data into an HTML <table>.
         The "text" field for a partitioned Table Element is always present, whether True or False.
+    keep_xml_tags
```
[the remark this answers is not in the document]
Comment (MthwRobinson) [reply unstructured/partition/auto.py]: Updated

File: docs/source/bricks.rst [hunk 3]
```diff
@@ -371,6 +371,26 @@ to disable SSL verification in the request.
...
 
+``partition_xml``
+-----------------
+
+The ``partition_xml`` function processes XML documents.
+If ``keep_xml_tags=False``, the function only returns the text attributes from the tags.
+You can use ``xml_path`` in conjuntion with ``keep_xml_tags=False`` to restrict the text
+extraction to specific tags.
+If ``keep_xml_tags=True``, the function returns tag information in addition to tag text.
+``keep_xml_tags`` is ``False`` be default.
```
[the remark this answers is not in the document]
Comment (MthwRobinson) [reply docs/source/bricks.rst]: Updated

**Pull request merged**